src/Plugin/AdvancedQueue/JobType/NetsEasyPaymentHandler.php

```diff
@@ -0,0 +1,298 @@
+   */
+  public function process(Job $job): JobResult {
+    $payload = $job->getPayload();
+    $stage = $payload['processing_stage'] ?? 0;
+
+    /** @var \Drupal\webform\Entity\WebformSubmissionInterface $webformSubmission */
+    $webformSubmission = WebformSubmission::load($payload['submissionId']);
+    $logger_context = [
+      'handler_id' => 'os2forms_payment',
+      'channel' => 'webform_submission',
+      'webform_submission' => $webformSubmission,
+      'operation' => 'response from queue',
+    ];
+
+    try {
+
+      if ($stage < 1) {
+        $this->getPaymentAndSetRelevantValues($job, $webformSubmission, $logger_context);
+      }
+
+      if ($stage < 2) {
+        $this->updatePaymentReference($job, $webformSubmission, $logger_context);
+      }
+
+      if ($stage < 3) {
+        $this->chargePayment($job, $webformSubmission, $logger_context);
+      }
+
+      $this->submissionLogger->notice($this->t('The submission #@serial was successfully delivered', ['@serial' => $webformSubmission->serial()]), $logger_context);
+
+      return JobResult::success();
+    }
+    catch (\Exception | GuzzleException $e) {
+      $this->submissionLogger->error($this->t('The submission #@serial failed (@message)', [
+        '@serial' => $webformSubmission->serial(),
+        '@message' => $e->getMessage(),
+      ]), $logger_context);
+
+      return JobResult::failure($e->getMessage());
+    }
+  }
+
+  /**
+   * Get payment information and set relevant values in the job payload.
+   *
+   * @param \Drupal\advancedqueue\Job $job
+   *   The job object containing payload information.
+   * @param \Drupal\webform\Entity\WebformSubmissionInterface $webformSubmission
+   *   The webform submission related to the payment.
+   * @param array $logger_context
+   *   Context for logging.
+   *
+   * @throws \Exception|GuzzleException
+   *   Throws Exception.
+   */
+  private function getPaymentAndSetRelevantValues(Job $job, WebformSubmissionInterface $webformSubmission, array $logger_context): void {
+    try {
+      $payload = $job->getPayload();
+      $paymentId = $payload['paymentId'];
+      $submissionId = $webformSubmission->id();
+      $webformId = $webformSubmission->getWebform()->id();
+
+      // Retrieve payment, save reference field.
+      $paymentEndpoint = $this->paymentHelper->getPaymentEndpoint($paymentId);
+      $result = $this->paymentHelper->handleApiRequest('GET', $paymentEndpoint);
+
+      $checkoutUrl = $result->payment->checkout->url;
+      $reservedAmount = floatval($result->payment->summary->reservedAmount ?? 0);
+      $chargedAmount = floatval($result->payment->summary->chargedAmount ?? 0);
+      $paymentReferenceSuffix = $result->payment->orderDetails->reference;
+
+      // Payment reference (field in NEXI backend)
+      // is defined as {optional suffix}:{webform_id}:{submission_id}.
+      $paymentReferenceValue = $webformId . ':' . $submissionId;
+
+      if ("undefined" !== $paymentReferenceSuffix) {
```

Comment:
Consistency is king!

```suggestion
      if ('undefined' !== $paymentReferenceSuffix) {
```